### PR TITLE
[staging] Update TinyMCE

### PR DESCRIPTION
**Pull request by brianteeman**
This PR updates tinymce to version 4.5.11 - this is the highest version we can include in Joomla 3 due to browser support

Version 4.5.11 (2019-05-16)
  Fixed bug where the editor would scroll to the top of the editable area if a dialog was closed in inline mode. #TINY-1073
Version 4.5.10 (2018-10-19)
  Changed the contextual toolbar shortcut to Ctrl+F9 since an Edge shortcut interfered with the previous one.

Updated references to website and company name

### CODE REVIEW

**Comment by brianteeman** [comment]
@zero-24 @SniperSister can you take a look at rips please
> Failure: Can't connect to the API

File: plugins/editors/tinymce/tinymce.xml [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-	<version>4.5.9</version>
-	<creationDate>2005-2018</creationDate>
-	<author>Ephox Corporation</author>
+	<version>4.5.11</version>
+	<creationDate>2005-2019</creationDate>
+	<author>Tiny Technologies, Inc</author>
 	<authorEmail>N/A</authorEmail>
-	<authorUrl>http://www.tinymce.com</authorUrl>
-	<copyright>Ephox Corporation</copyright>
+	<authorUrl>http://www.tiny.cloud</authorUrl>
```
Comment (Quy) [inline plugins/editors/tinymce/tinymce.xml]:
```suggestion
	<authorUrl>https://www.tiny.cloud</authorUrl>
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 3f982192851743e01d058fcb9defd3ec01df6ef2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24978">issues.joomla.org/tracker/joomla-cms/24978</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on dabc0f6d392e91cb64140c4b5ce0bf88e8bb5bd0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24978">issues.joomla.org/tracker/joomla-cms/24978</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24978">issues.joomla.org/tracker/joomla-cms/24978</a>.</sub>

**Comment by HLeithner** [comment]
thx

**Comment by HLeithner** [comment]
thx

**Comment by brianteeman** [comment]
thanks

**Comment by BurtNL** [comment]
Could the introduction of this new version of Tiny MCE cause issues in modules that are created by third parties? I am using third party modules in which HTML markup is used, in textarea fields. Until J 3.9.6 this was no problem, but in J 3.9.8 all HTML markup is stripped out. Besides the update to J 3.9.8 no other settings were changed regarding Text Filtering and so on.
I have looked at the settings in the Tiny MCE plugin and tried different settings, without good result, the HTML markup gets stripped out of all fields in the third party module.

**Comment by brianteeman** [comment]
No this would have no impact

**Comment by HLeithner** [comment]
There is an issue with custom subfields and filtering html from textarea and edtor fields.
This is already fixed for the next version.

Do you mean this?

**Comment by BurtNL** [comment]
J396
J398

Hope the screenshots makes clear what I mean. In J3.9.6 HTML is allowed and saved properly.
After update to J3.9.8 exactly the same module gets stripped out of HTML.
As said before, I haven't changed or updated anything else besides the Joomla version.

**Comment by BurtNL** [comment]
I know it is not core Joomla, but I would not have expected this.

**Comment by HLeithner** [comment]
This problem is fixed in https://github.com/joomla/joomla-cms/pull/25189 it will be fixed with the next version. In the meantime you could apply the patch.

**Comment by BurtNL** [comment]
I have downloaded both files from #25189 and replaced them in my local Joomla setup, but it does not solve the issue. The HTML is still stripped. For Custom module (core) is works good, but for the third party modules not.

**Comment by BurtNL** [comment]
I will download staging tomorrow and give it try again.

**Comment by BurtNL** [comment]
Just downloaded and installed J 3.9.9-dev, but still seeing the same issue.

J399-dev


After entering <p>Training</p> and saving only the text is left, HTML is gone.
In J 3.9.6 not an issue.

**Comment by HLeithner** [comment]
@relater could you please check if its releated to your PR and if we can fix it?

**Comment by ReLater** [comment]
It's not related to my pr directly. I just provided a fix for a similiar issue for subform fields in com_fields::repeatable. And the issue here is not related to TinyMCE

**All extensions that use subform fields MUST add an attribute `filter` to their subform child fields of type `editor`, `textarea`, `text` (maybe others, too) since Joomla 3.9.7 like it's common for "normal" JForm fields if you want to allow HTML input. Otherwise the validation falls back to `STRING`, which is the common behavior for "normal" JForm fields.**

e.g. 
`filter="safehtml"`
`filter="JComponentHelper::filterText"`
`filter="raw"` (bad decision in most cases)

Reason is a **SECURITY** fix in Joomla 3.9.7. Subform child fields weren't validated before that fix and  for example JavaScript injections were possible without any cleaning.

Before that fix filter attributes in subform child fields were completely effectless!!

Example from core for a "good" field declaration:
```
<field
	name="tag_list_description"
	type="textarea"
	class="inputbox"
	label="COM_TAGS_SHOW_TAG_LIST_DESCRIPTION_LABEL"
	description="COM_TAGS_TAG_LIST_DESCRIPTION_DESC"
	rows="3"
	cols="30"
	filter="safehtml"
/>
```

**Comment by ReLater** [comment]
FYI: Added info to docs: https://docs.joomla.org/Subform_form_field_type#Beware.21

Maybe it should be propagated at other locations, too.

**Comment by HLeithner** [comment]
Thx @ReLater for the confirmation.

@BurtNL can you please contact the author of the extension und notify him/her of this change?

**Comment by BurtNL** [comment]
Thanks @ReLater and @HLeithner , I have informed the author of this module already and send him the links to this thread and the docs.

**Pull request merged**